test/unittests/t_catalog_mgr.cc

```diff
@@ -0,0 +1,174 @@
+    MockCatalog *root_catalog = catalog_mgr_->RetrieveRootCatalog();
+    root_catalog->AddFile(empty_content, file_size, "", "");
+    // adding "/dir"
+    root_catalog->AddFile(empty_content, file_size, "", "dir");
+    // adding "/file1"
+    hash = shash::Any(shash::kSha1,
+                      reinterpret_cast<const unsigned char*>(hashes[1]),
+                      size, suffix);
+    root_catalog->AddFile(hash, file_size, "", "file1");
+    // adding "/dir/dir"
+    root_catalog->AddFile(empty_content, file_size, "/dir", "dir");
+    // adding "/dir/dir/file2"
+    hash = shash::Any(shash::kSha1,
+                      reinterpret_cast<const unsigned char*>(hashes[1]),
+                      size, suffix);
+    root_catalog->AddFile(hash, file_size, "/dir/dir", "file2");
+    // adding "/dir/dir/dir"
+    root_catalog->AddFile(empty_content, file_size, "/dir/dir", "dir");
+    // adding a nested catalog in "/dir/dir/dir"
+    string catalog_path_str = "/dir/dir/dir";
+    PathString catalog_path(catalog_path_str.c_str(),
+                            catalog_path_str.length());
+    MockCatalog *new_catalog = new MockCatalog("/dir/dir/dir", shash::Any(),
+                                               4096, 1, 0, false,
+                                               root_catalog, NULL);
+    ASSERT_NE(static_cast<MockCatalog*>(NULL), new_catalog);
+    ASSERT_EQ(1u, root_catalog->GetChildren().size());
+    catalog_mgr_->RegisterNewCatalog(new_catalog);
+    // adding "/dir/dir/dir/file3" to the new nested catalog
+    hash = shash::Any(shash::kSha1,
+                          reinterpret_cast<const unsigned char*>(hashes[2]),
+                          size, suffix);
+    new_catalog->AddFile(hash, file_size, "/dir/dir/dir", "file3");
+    // adding "/dir/dir/dir/file4" to the new nested catalog
+    hash = shash::Any(shash::kSha1,
+                          reinterpret_cast<const unsigned char*>(hashes[3]),
+                          size, suffix);
+    new_catalog->AddFile(hash, file_size, "/dir/dir/dir", "file4");
+    // we haven't mounted the second catalog yet!
+    ASSERT_EQ(1, catalog_mgr_->GetNumCatalogs());
+  }
+
+ protected:
+  const static char *hashes[];
+  MockCatalogManager *catalog_mgr_;
+  perf::Statistics statistics_;
+};
+
+const char *T_CatalogManager::hashes[] = {
+     "b026324c6904b2a9cb4b88d6d61c81d1000000",
+     "26ab0db90d72e28ad0ba1e22ee510510000000",
+     "6d7fce9fee471194aa8b5b6e47267f03000000",
+     "48a24b70a0b376535542b996af517398000000",
+     "1dcca23355272056f04fe8bf20edfce0000000"
+};
+
+
+
+TEST_F(T_CatalogManager, InitialConfiguration) {
+  EXPECT_TRUE(catalog_mgr_->Init());
+  EXPECT_EQ(1, catalog_mgr_->GetNumCatalogs());
+  EXPECT_EQ(1u, catalog_mgr_->GetRevision());
+  EXPECT_FALSE(catalog_mgr_->GetVolatileFlag());
+  EXPECT_EQ(0u, catalog_mgr_->GetTTL());
+}
+
+TEST_F(T_CatalogManager, Statistics) {
+  EXPECT_TRUE(catalog_mgr_->Init());
+  Statistics st = catalog_mgr_->statistics();
```

Comment:
Could actually be a `const Statistics&` both here and in the `CatalogManager` to avoid copying.